syntax="proto3";
//option cc_generic_services = true;

package ccraft.protocal.serverraft;

message RfLogEntry {
    uint32 term         = 1;
    uint64 index        = 2;
    uint32 type         = 3;
    bytes  data         = 4;
}

message AppendRfLogRequest {
    uint32                term         = 1;
    uint32                leaderId     = 2;
    uint64                prevLogIndex = 3;
    uint32                prevLogTerm  = 4;
    repeated RfLogEntry   entries      = 5;
    uint64                leaderCommit = 6;
}

message AppendRfLogResponse {
    uint32 term    = 1;
    bool   success = 2;
}

message RequestVoteRequest {
    uint32                term         = 1;
    uint32                leaderId     = 2;
    uint64                lastLogIndex = 3;
    uint32                lastLogTerm  = 4;
}

message RequestVoteResponse {
    uint32 term = 1;
    bool   success = 2;
}
